test/unit/utils/UWebSocketTransport.spec.ts

```diff
@@ -10,270 +15,693 @@ describe('UWebSocketTransport', () => {
+      await jest.runAllTimersAsync()
+    })
+  })
+
+  describe('Message Processing', () => {
+    it('should handle incoming binary messages correctly', () => {
+      const messageListener = jest.fn()
+      transport.on('message', messageListener)
+
+      const buffer = new ArrayBuffer(4)
+      mockEmitter.emit('message', buffer)
+
+      expect(messageListener).toHaveBeenCalledWith(new Uint8Array(buffer))
+    })
+
+    it('should reject non-binary messages with appropriate error', () => {
+      mockEmitter.emit('message', 'invalid message')
+
+      expect(errorListener).toHaveBeenCalledWith(new Error('WebSocketTransport: Received unknown type of message'))
+    })
+
+    it('should handle message emission errors gracefully', () => {
+      const error = new Error('Emission error')
+      transport.on('message', () => {
+        throw error
+      })
+
+      mockEmitter.emit('message', new ArrayBuffer(4))
+
+      expect(errorListener).toHaveBeenCalledWith(new Error(`Failed to emit message: ${error.message}`))
     })
   })
 
   describe('Error Handling', () => {
-    it('should handle socket errors during send', async () => {
+    it('should handle socket send errors with proper error propagation and logging', async () => {
       const error = new Error('Socket error')
       mockSocket.send.mockImplementation(() => {
         throw error
       })
 
-      const message = new Uint8Array([1, 2, 3])
-      await expect(transport.sendMessage(message)).rejects.toThrow('Failed to send message: Socket error')
+      const message = new Uint8Array([1])
+      const sendPromise = transport.sendMessage(message)
+
+      // Add catch after assertion to prevent Jest unhandled rejection error
+      ;(sendPromise as unknown as Promise<void>)?.catch(() => {})
+
+      // Process the attempt
+      await jest.advanceTimersByTimeAsync(0)
+
+      await expect(sendPromise).rejects.toThrow('Failed to send message: Socket error')
+
+      // Check error log format (exact message and payload structure)
+      expect(mockLogs.getLogger('ws-transport').error).toHaveBeenCalledWith(
+        'Error sending message', // Exact message from code
+        {
+          transportId: expect.any(String),
+          error: 'Failed to send message: Socket error'
+        }
+      )
+
+      // Check transport error event emission
+      expect(errorListener).toHaveBeenCalledWith(new Error('Failed to send message: Socket error'))
     })
 
-    it('should handle invalid message types', () => {
+    it('should handle invalid message types with type checking', () => {
       transport.sendMessage('invalid' as any)
+
       expect(errorListener).toHaveBeenCalledWith(
         new Error('WebSocketTransport: Received unknown type of message, expecting Uint8Array')
       )
     })
+
+    it('should handle cleanup errors gracefully', () => {
+      const error = new Error('Cleanup error')
+      mockSocket.end.mockImplementation(() => {
```

Comment:
If I'm not wrong, when sending data, the code is not using end, should we test this?